src/dialog-utilities.js

```diff
@@ -0,0 +1,39 @@
+import {Origin} from 'aurelia-metadata';
+
+/**
+* Returns a view model based on a given instruction
+*
+* @param {Object} instruction the instruction from which the view model is determined
+* @return {Promise} the view model
+*/
+export function getViewModel(instruction: any, compositionEngine: CompositionEngine) {
+  if (typeof instruction.viewModel === 'function') {
+    instruction.viewModel = Origin.get(instruction.viewModel).moduleId;
+  } else if (typeof instruction.viewModel === 'string') {
+    return compositionEngine.ensureViewModel(instruction);
+  }
+
+  return Promise.resolve(instruction);
+}
+
+/**
+* Invokes a lifecycle event on a given instance
+*
+* @param {Object} instance  the instance to invoke the event on
+* @param {String} name      the event to invoke
+* @param {Object} [model]   the optional model to invoke the event with
+* @return {Promise} the result of the invocation
+*/
+export function invokeLifecycle(instance: any, name: string, model: any) {
```

Comment:
Typically we try to add test coverage as things are refactored, especially something like this where we move a utility in to a separate file from where it's used.  It helps to document the code through tests also so the next person who wants to submit a PR can get a better understanding of what each piece does and how it works together.

If you can document the behavior through failing specs at least I can help to get them passing.